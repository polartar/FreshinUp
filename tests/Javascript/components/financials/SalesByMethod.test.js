import { mount, createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Component from '~/components/financials/SalesByMethod.vue'
import { FIXTURE_PAYMENT_TYPE_TOTALS } from 'tests/__data__/paymentTypesTotal'

describe('financials/SalesByMethod', () => {
  let localVue
  describe('Snapshots', () => {
    beforeEach(() => {
      localVue = createLocalVue()
      localVue.use(Vuetify)
    })
    test('sales set', () => {
      const component = mount(Component, {
        propsData: {
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        },
        localVue,
        stubs: {
          Doughnut: true
        }
      })
      expect(component.element).toMatchSnapshot()
    })
  })
  describe('Computed', () => {
    test('values', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        },
        stubs: {
          Doughnut: true
        }
      })
      expect(wrapper.vm.values).toEqual([30000, 40000, 35000, 20000])
    })
    test('labels', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        },
        stubs: {
          Doughnut: true
        }
      })
      expect(wrapper.vm.labels).toEqual(['VISA 24%', 'MASTERCARD 32%', 'AMEX 28%', 'CASH 16%'])
    })
    test('total', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        },
        stubs: {
          Doughnut: true
        }
      })
      expect(wrapper.vm.total).toEqual(125000)
    })
    test('chartData', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        },
        stubs: {
          Lines: true
        }
      })
      expect(wrapper.vm.chartData).toEqual({ 'datasets': [{ 'backgroundColor': ['#1976D2', '#FB8C00', '#424242', '#2196F3'], 'borderWidth': 0, 'data': wrapper.vm.values }], 'labels': wrapper.vm.labels })
    })
  })
  describe('Methods', () => {
    test('getPercentageString', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        },
        stubs: {
          Doughnut: true
        }
      })
      expect(wrapper.vm.getPercentageString(20000)).toEqual('16%')
    })
  })
})
