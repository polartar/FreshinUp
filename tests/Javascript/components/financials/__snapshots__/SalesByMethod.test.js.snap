// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`financials/SalesByMethod Snapshots sales set 1`] = `
<div
  class="container"
>
  <div
    class="layout row wrap align-center"
  >
    <div
      class="flex xs12 sm6"
    >
      <doughnut-stub
        chartdata="[object Object]"
        chartid="doughnut-chart"
        cssclasses=""
        height="400"
        options="[object Object]"
        plugins=""
        width="400"
      />
    </div>
     
    <div
      class="flex xs12 sm6 pl-4"
    >
      <div
        class="layout d-flex mb-3 row align-left"
      >
        <div
          class="flex"
        >
          <i
            aria-hidden="true"
            class="v-icon mr-2 fa fa-circle theme--light"
            style="font-size: 16px; color: rgb(25, 118, 210); caret-color: #1976D2;"
          />
           
          <span
            class="body-2 text-lowercase"
          >
            VISA 24%
          </span>
        </div>
      </div>
      <div
        class="layout d-flex mb-3 row align-left"
      >
        <div
          class="flex"
        >
          <i
            aria-hidden="true"
            class="v-icon mr-2 fa fa-circle theme--light"
            style="font-size: 16px; color: rgb(251, 140, 0); caret-color: #FB8C00;"
          />
           
          <span
            class="body-2 text-lowercase"
          >
            MASTERCARD 32%
          </span>
        </div>
      </div>
      <div
        class="layout d-flex mb-3 row align-left"
      >
        <div
          class="flex"
        >
          <i
            aria-hidden="true"
            class="v-icon mr-2 fa fa-circle theme--light"
            style="font-size: 16px; color: rgb(66, 66, 66); caret-color: #424242;"
          />
           
          <span
            class="body-2 text-lowercase"
          >
            AMEX 28%
          </span>
        </div>
      </div>
      <div
        class="layout d-flex mb-3 row align-left"
      >
        <div
          class="flex"
        >
          <i
            aria-hidden="true"
            class="v-icon mr-2 fa fa-circle theme--light"
            style="font-size: 16px; color: rgb(33, 150, 243); caret-color: #2196F3;"
          />
           
          <span
            class="body-2 text-lowercase"
          >
            CASH 16%
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
